hcl/fmt.go

```diff
@@ -210,62 +194,67 @@ func fmtAttrExpr(tokens hclwrite.Tokens) hclwrite.Tokens {
 			tokensStr(tokens),
 		))
 	}
-	return formattedList
+	return formatted
 }
 
 // fmtListExpr will adjust the given list tokens so they can be formatted
 // properly. It returns the adjusted tokens and the position of the first
 // token after the list ended.
 //
+// isInsideObject is necessary to disambiguate between a list indexing operation
+// and lists being used as keys inside an object.
```

Comment:
Terraform doesn't even allowthis, but it is allowed on HCL, so I tried to cover for it..but it is tricky because of the different ways that indexing is allowed, with newlines between them etc... like this is valid:

```
a = []
[0]
```

But would be very similar to what would happen inside an object. So far it seems this logic is relativelly good enough...still fuzzing.